components/avatar/avatar.vue

```diff
@@ -194,31 +203,76 @@ export default {
       type: [String, Array, Object],
       default: '',
     },
+
+    /**
+     * Source of the image
+     */
+    imageSrc: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Alt attribute of the image, by default
+     * it'd be the full name
+     */
+    imageAlt: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Icon name to be displayed on the avatar
+     */
+    iconName: {
+      type: String,
+      default: undefined,
+      validator: (name) => ICONS_LIST.includes(name),
+    },
+
+    /**
+     * Icon size to be displayed on the avatar
+     * @values 100, 200, 300, 400, 500, 600, 700, 800
+     */
+    iconSize: {
+      type: String,
+      default: '500',
+      validator: (size) => Object.keys(ICON_SIZE_MODIFIERS).includes(size),
+    },
+
+    /**
+     * Full name used to extract initials and set alt attribute.
+     */
+    fullName: {
+      type: String,
+      default: '',
+    },
   },
 
   data () {
     return {
-      // initials, image or icon
-      kind: null,
       AVATAR_SIZE_MODIFIERS,
       AVATAR_KIND_MODIFIERS,
       AVATAR_PRESENCE_SIZE_MODIFIERS,
       imageLoadedSuccessfully: null,
-      slottedInitials: '',
       formattedInitials: '',
       initializing: false,
     };
   },
 
   computed: {
+    isIconType () {
+      return !!this.iconName;
+    },
+
     avatarClasses () {
       return [
         'd-avatar',
         AVATAR_SIZE_MODIFIERS[this.validatedSize],
         this.avatarClass,
         {
           'd-avatar--group': this.showGroup,
-          [`d-avatar--color-${this.getColor()}`]: this.kind !== 'icon',
+          [`d-avatar--color-${this.getColor()}`]: !this.isIconType,
```

Comment:
Since `isIconType` seems is a computed only for this 
```[`d-avatar--color-${this.getColor()}`]: !this.isIconType,```

II suggest the computed name is `isNotIconType` and returns `!this.iconName;`